src/lib/operators/join_sort_merge.cpp

```diff
@@ -893,6 +1223,49 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
         _output_pos_lists_right.push_back(std::move(null_output_right));
       }
     }
+
+    if (_mode == JoinMode::AntiNullAsTrue) {
+      // If the join mode is anti null as true with secondary predicates, we need to check for every left null row that:
+      // 1. There is no match with one of the right null rows where all predicates are satisfied.
+      // 2. There is no match with one of the left rows (not null) where all predicates are satisfied.
```

Comment:
I don't understand. We compare a left NULL value with another left non-NULL value?

Reply:
It's a typo. It should be right non-NULL.

Reply:
Does it make now sence?